frontend/src/utils/api.ts

```diff
@@ -290,12 +303,14 @@ window.api = {
   setContributionOfCommitResultsAndInsertRepoId(dailyCommits, repoId) {
     dailyCommits.forEach((commit) => {
       commit.commitResults.forEach((result) => {
-        result.repoId = repoId;
-        result.insertions = Object.values(result.fileTypesAndContributionMap)
+        (result as CommitResult).repoId = repoId;
+        (result as CommitResult).insertions = Object.values(result.fileTypesAndContributionMap)
             .reduce((acc, fileType) => acc + fileType.insertions, 0);
-        result.deletions = Object.values(result.fileTypesAndContributionMap)
+        (result as CommitResult).deletions = Object.values(result.fileTypesAndContributionMap)
             .reduce((acc, fileType) => acc + fileType.deletions, 0);
       });
+      // typecast to DailyCommit since we added repoId/insertions/deletions
+      commit as any as DailyCommit;
```

Comment:
```suggestion
      commit as unknown as DailyCommit;
```
Is it better to typecast to `unknown` instead of `any` here?

Reply:
Actually I realised that this typecast wasn't doing anything 😅 since it doesn't change the underlying type of the object, but creates a new object with the new type.
I originally wanted to typecast this to have the correct type when creating the `User` object. For now the `obj` in `new User(obj)` has the `any` type - changing this will require some refactoring which I'll do in a separate PR as mentioned above.